src/application.ts

```diff
@@ -105,6 +131,30 @@ export default class Application implements Owner {
    * Owner interface implementation
    */
 
+  register(specifier: string, factory: FactoryDefinition<any>, options?: RegistrationOptions) {
+    if (this._initialized) {
+      throw new Error("You can't add new registrations after an application has booted. Use an initializer instead.");
+    }
+
+    this._registry.register(specifier, factory, options);
+  }
+
+  unregister(specifier: string) {
+    this._registry.unregister(specifier);
+  }
+
+  registerOption(specifier: string, option: string, value: any): void {
+    this._registry.registerOption(specifier, option, value);
+  }
+
+  unregisterOption(specifier: string, option: string): void {
+    this._registry.unregisterOption(specifier, option);
+  }
+
+  registerInjection(specifier: string, property: string, source: string): void {
+    this._registry.registerInjection(specifier, property, source);
+  }
+
```

Comment:
I am concerned about any registry mutations after initialization and would prefer to leave these proxy methods off the `Application` entirely. They will only encourage ad hoc registrations in JS-land, such as `getOwner(this).register()`. I think we'll be much better off if the registry can be considered "sealed" and read-only after initialization.